### DISCUSSION TITLE
mypy issues when using run() for a Task

### DISCUSSION THREAD

**Discussion by isaacna** [PrefectHQ/prefect#3896] [Q&A] [answered]
I'm trying to unit test tasks in the following format (as suggested [here](https://github.com/PrefectHQ/prefect/issues/2627)):

```
@task
def plus_one(x):
    return x + 1


def test_plus_one():
    assert 3 == plus_one.run(2)
```

This is working as expected, but I run into the following errors when running `mypy` on the test file:
```
error: Too many arguments for "run" of "Task"
error: "run" of "Task" does not return a value
```

Could this be because I am using the task decorator rather than subclassing `Task` and creating a `run` method? And is there a workaround to be able to test this way?

**Comment by cicdw** [comment, marked as answer]
Hi @isaacna unfortunately at this time `mypy` does not play well with Prefect tasks, and you'd see those mypy errors regardless of whether you use `@task` or create a subclass yourself.

For more info, check out this thread: https://github.com/PrefectHQ/prefect/issues/3191

**Reply by isaacna** [reply]
Ah gotcha, thanks! I guess for now I'll just ignore those lines with `# type: ignore`
